app/src/main/java/it/quip/android/actvitiy/QuipitHomeActivity.java

```diff
@@ -112,20 +145,44 @@ public boolean onOptionsItemSelected(MenuItem item) {
             case android.R.id.home:
                 dlDrawer.openDrawer(GravityCompat.START);
                 return true;
+            case R.id.miNewCircle:
+                createCircle();
+                return true;
         }
 
         return super.onOptionsItemSelected(item);
     }
 
     @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (CREATE_CIRCLE_REQUEST == requestCode) {
+            if (RESULT_OK == resultCode) {
+                Circle createdCircle = data.getParcelableExtra(CreateCircleActivity.CREATED_CIRCLE);
+                onCircleCreated(createdCircle);
+            }
+        }
+    }
+
+    private void createCircle() {
+        Intent intent = new Intent(this, CreateCircleActivity.class);
+        startActivityForResult(intent, CREATE_CIRCLE_REQUEST);
+    }
+
+    private void onCircleCreated(Circle createdCircle) {
+        user.addCircle(createdCircle);
+        updateSidebarMenu();
+        prepareFragment(ViewCircleFragment.newInstance(createdCircle)).commitAllowingStateLoss();
```

Comment:
I'm going to put a comment here. If you use commit after onSaveInstanceState or whatever, its an illegal state. It saves the state when you start a new activity from this one. So in this case, since we're coming back from an activity on the result, we want to swap the fragment but say we know what we're doing

The way I understand it, it doesn't save the state such that if someone closes the app and reopens, it might not look like they'd expect. In this case, its ok since we can hop back to the quips feed